pkgs/desktops/gnome-3/core/gnome-session/fix-paths.patch

```diff
@@ -20,3 +36,42 @@
                  g_set_error (error, 
                               G_SPAWN_ERROR,
                               G_SPAWN_ERROR_FAILED,
+--- a/gnome-session/gsm-manager.c
++++ b/gnome-session/gsm-manager.c
+@@ -3192,7 +3192,9 @@ gsm_manager_init (GsmManager *manager)
+ 
+         manager->priv = GSM_MANAGER_GET_PRIVATE (manager);
+ 
+-        manager->priv->settings = g_settings_new (GSM_MANAGER_SCHEMA);
++        GSettingsSchemaSource *schema_source = g_settings_schema_source_new_from_directory("@gschemasCompiled@", NULL, FALSE, NULL);
```

Comment:
`GSettingsSchemaSource` and `GSettingsSchema` need to be freed using [`g_settings_schema_source_unref`](https://developer.gnome.org/gio/stable/gio-GSettingsSchema-GSettingsSchemaSource.html#g-settings-schema-source-unref) and [`g_settings_schema_unref`](https://developer.gnome.org/gio/stable/gio-GSettingsSchema-GSettingsSchemaSource.html#g-settings-schema-unref) respectively.

Reply:
Tried freeing right after `manager->priv->settings` is assigned, and that seemed to work. Though I can see that you've waited to free until the settings object is freed in the gpaste patch, so I'm not entirely sure if it actually works.

If `settings` hold a reference to the `schema` and `schema_source` I'm guessing it will correctly free it when destroyed?

Reply:
They are reference counted so if `settings` holds the reference it will unref it, otherwise there should be no issue.

Reply:
The VM worked at least, so going with that.

Reply:
Yeah, got a confirmation on IRC.

* https://gitlab.gnome.org/GNOME/glib/blob/516101c7023af25083c385b91a1ca7dc635c15a6/gio/gsettingsschema.c#L177
* https://gitlab.gnome.org/GNOME/glib/blob/516101c7023af25083c385b91a1ca7dc635c15a6/gio/gsettings.c#L548
* https://gitlab.gnome.org/GNOME/glib/blob/516101c7023af25083c385b91a1ca7dc635c15a6/gio/gsettings.c#L716